bundles/http/Citrix/AdobeSign/31fff7a5-ffa8-4019-9e2f-e477837f92de.js

```diff
@@ -0,0 +1,596 @@
+                "esign": Boolean(agreement.esign),
+                "type": agreement.type,
+                "name": agreement.name,
+                "latestVersionId": agreement.latestVersionId,
+                "status": getStatus(agreement.status),
+                "hidden": Boolean(agreement.hidden),
+                "groupId": agreement.groupId,
+                "userId": userids[i]
+            }
+        })
+
+        dataStore.save("agreements", agreementsData ?? []);
+        i++
+    } while (i < userids.length)
+    return agreements
+}
+
+async function syncLibraryDocuments(dataStore, client, userids) {
+    const PAGE_SIZE = 100
+    let cursor = " "
+    let i = 0
+    if (userids.length == 0) {
+        return false
+    }
+    let url = `/api/rest/v6/libraryDocuments?pageSize=${PAGE_SIZE}`
+    do {
+        const response = await client.fetch(url, {
+            headers: {
+                "x-api-user": `userid:${userids[i]}`,
+            }
+        })
+        const json = await response.json()
+        if (!response.ok) {
+            throw new Error(JSON.stringify(json))
+        }
+        cursor = json.page.nextCursor ?? null
+        url = `/api/rest/v6/libraryDocuments?pageSize=${PAGE_SIZE}&cursor=${cursor}`
+        let templateData = json.libraryDocumentList?.map(template => {
+            let templateTypesData = template.templateTypes?.map(templateType => {
+                return {
+                    "templateType": templateType,
+                    "parentId": template.id,
+                    "uniqueId": uuid.v4()
+                }
+            })
+            dataStore.save('templateTypes', templateTypesData ?? [])
+            return {
+                "id": template.id,
+                "name": template.name,
+                "creatorEmail": template.creatorEmail,
+                "ownerEmail": template.ownerEmail,
+                "modifiedDate": new Date(template.modifiedDate),
+                "sharingMode": template.sharingMode,
+                "hidden": Boolean(template.hidden),
+                "groupId": template.groupId,
+                "status": getStatus(template.status)
+            }
+        })
+
+        dataStore.save("libraryDocuments", templateData ?? []);
+
+        if (cursor == null) {
+            i++
+            url = `/api/rest/v6/libraryDocuments?page_size=${PAGE_SIZE}`
+        }
+
+    } while (i < userids.length)
+}
+
+async function syncArgreementDetails(client, dataStore, agreements) {
+    for (const [agreementId, value] of agreements) {
+        const response = await client.fetch(`/api/rest/v6/agreements/${agreementId}/members`, {
+            headers: {
+                "x-api-user": `userid:${value.userid}`,
+            }
+        })
+        const json = await response.json()
+        if (!response.ok) {
+            throw new Error(JSON.stringify(json))
+        }
+        json.participantSets?.map(participantSet => {
```

Comment:
```suggestion
        json.participantSets?.forEach(participantSet => {
```
You don't need to create a new array

Reply:
changes implemented